sw/host/tests/manuf/manuf_cp_test_lock/src/main.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This test performs a lifecycle transition from `TEST_UNLOCKED0` to `TEST_LOCKED0`.
+//!
+//! We also test provisioning a `TEST_UNLOCKED` being programmed into OTP.
+
+use std::rc::Rc;
+
+use anyhow::Context;
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::dif::lc_ctrl::{DifLcCtrlState, LcCtrlReg};
+use opentitanlib::dif::otp_ctrl::DamParam;
+use opentitanlib::execute_test;
+use opentitanlib::io::jtag::{Jtag, JtagParams, JtagTap};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::lc_transition;
+use opentitanlib::test_utils::otp_ctrl::OtpParam;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams,
+}
+
+/// Pre-image of the TEST_UNLOCK token that will be written to OTP.
+const TEST_UNLOCK_TOKEN_PREIMAGE: [u32; 4] = [0xAAAAAAAA, 0xAAAAAAAA, 0xAAAAAAAA, 0xAAAAAAAA];
+/// Post-image of the TEST_UNLOCK token as generated by `gen_test_unlock_token.py`.
+const TEST_UNLOCK_TOKEN_POSTIMAGE: [u32; 4] = [0xe6b9ec3f, 0x67fd75f3, 0x618ae065, 0xc0bb8f81];
+
+fn main() -> anyhow::Result<()> {
+    let opts = Opts::from_args();
+    opts.init.init_logging();
+
+    let transport = opts.init.init_target()?;
+
+    execute_test!(provision_token, &opts, &transport);
+    execute_test!(test_lock, &opts, &transport);
+    execute_test!(test_unlock, &opts, &transport);
+
+    Ok(())
+}
+
+/// Provision the `TEST_UNLOCK` token, programming it into OTP.
+fn provision_token(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let jtag = reset_to_tap(opts, transport, JtagTap::RiscvTap)?;
+
+    OtpParam::write_param(
+        &*jtag,
+        DamParam::TestUnlockToken,
+        &TEST_UNLOCK_TOKEN_POSTIMAGE,
+    )
+    .context("failed to write TEST_UNLOCK token to OTP")?;
+
+    // Read back the token to verify it was written.
+    let mut test_unlock_token = [0xff; 4];
+    OtpParam::read_param(&*jtag, DamParam::TestUnlockToken, &mut test_unlock_token)
+        .context("failed to read back TEST_UNLOCK token from OTP")?;
```

Comment:
I believe you need to lock this partition, and trigger a reset, before the token will become active. Though I can't seem to locate this in the docs.

@msfschaffner : for the secret0 partition, do you need to lock it / perform a reset to complete the locking after provisioning the TEST_UNLOCK/EXIT tokens for them to be ungated to the LC ctrl? Can't seem to find this mentioned in the docs.

Reply:
I've updated the `provision_token` step to lock `SECRET0` and reset after provisioning, but the LC controller is still giving a status with `TOKEN_ERROR` set when trying to unlock.

Do I need to make `TEST_EXIT` non-zero before locking or anything like that?

Reply:
In the `manuf_cp_device_flash_info_wr` test I set the test exit and unlock tokens to the same value, so you could just do that